build_bundle.py

```diff
@@ -0,0 +1,367 @@
+        print("create ostree repo")
+        os.mkdir(repo_path)
+        os.system("ostree --repo={} init --mode={}".format(
+            repo_path, OSTREE_REPO_TYPE))
+
+    def pull_rootfs(self, ostree_rootfs):
+        print("pull_rootfs")
+        ostree_commit = subprocess.check_output(
+            'ostree --repo={} rev-parse {}'.format(self.__repo_path,
+                                                   self.__branch),
+            shell=True, encoding='utf-8')
+        ostree_commit = ostree_commit.replace('\n', ' ').replace('\r', '')
+        if "status 1" in ostree_commit:
+            raise Exception("no initial commit befor incremental")
+        print('ostree commit = {}'.format(ostree_commit))
+        os.system("ostree --repo={}/.ostree_repo init --mode=bare-user".format(
+            ostree_rootfs))
+        os.system("ostree --repo={}/.ostree_repo pull-local {} {}".format(
+            ostree_rootfs, self.__repo_path, ostree_commit))
+        os.system(
+            "ostree --repo={}/.ostree_repo refs --create=aos_branch {}".format(
+                ostree_rootfs, ostree_commit))
+        os.system("ostree --repo={}/.ostree_repo checkout {}".format(
+            ostree_rootfs, ostree_commit))
+
+        #Removing ostree repo
+        shutil.rmtree('{}/.ostree_repo'.format(ostree_rootfs))
+
+    def commit_rootfs(self, rootfs_tar):
+        print("commit rootfs")
+        os.system(
+            'ostree --repo={} commit --tree=tar={} --skip-if-unchanged --branch={} --subject="{}"'.
+            format(self.__repo_path, rootfs_tar, self.__branch, component))
+
+
+MKSDCARD_IMAGE_PATH = "meta-xt-prod-aos/doc"
+
+
+class ImageBuilder:
+    def __init__(self, cfg):
+        print("image_builder")
+        self.__cfg = cfg
+        self.__temp = tempfile.mkdtemp()
+        self.__mksdcard_path = os.path.join(
+            self.__cfg.get(SECT_PATH, OPT_BASE), MKSDCARD_IMAGE_PATH)
+        self.__loop_dev = ""
+
+    def cleanup(self):
+        if self.__loop_dev:
+            os.system("sudo losetup -d {}".format(self.__loop_dev))
+        shutil.rmtree(self.__temp, ignore_errors=True)
+
+    def build_image(self, deploy_path):
+        print("building image to {}".format(deploy_path))
+        os.system("{}/mk_sdcard_image.sh -p {} -d {}/image.img -c aos".format(
+            self.__mksdcard_path, deploy_path, self.__temp))
+
+    def connect_image(self):
+        print("connect image")
+        self.__loop_dev = subprocess.check_output(
+            "sudo losetup --find --partscan --show {}/image.img".format(
+                self.__temp),
+            shell=True, encoding='utf-8')
+        print(self.__loop_dev)
+        self.__loop_dev = self.__loop_dev.replace('\n', '').replace('\r', '')
+        print("Got loop dev = {}".format(self.__loop_dev))
+
+    def get_partition_device(self, partno):
+        print("get part device {}".format(partno))
+        if not self.__loop_dev:
+            raise Exception("no loop device found")
```

Comment:
Don't use the `Exception` class. Use appropriate exception class (for example, own class or `RuntimeError`)